### DISCUSSION TITLE
Is usePrepareContractWrite still recommended in v1.x.x?

### DISCUSSION THREAD

**Discussion by DannyDelott** [wevm/wagmi#2826] [Q&A]
I noticed in the [docs](https://wagmi.sh/react/hooks/useContractWrite.en-US) that it's recommended to use usePrepareContractWrite with useContractWrite to avoid UX pitfalls. However, it doesn't appear in the navigation and in the [1.x.x migration guide](https://wagmi.sh/react/migration-guide#usecontractwrite:~:text=Removed%20requirement%20for%20%22prepare%22%20mode-,Removed%20the%20requirement%20for%20%22prepare%22%20mode.%20You%20can%20now%20use%20useContractWrite%20without%20calling%20usePrepareContractWrite%20first.,-const%20%7B%20data%2C%20isLoading%2C%20isSuccess%2C%20write), it says the "prepare" mode is no longer required.

Is usePrepareContractWrite still recommended in v1.x.x, or are we good to go without it?

**Comment by tmm** [comment]
Everything listed [here](https://wagmi.sh/react/prepare-hooks#ux-pitfalls-without-prepare-hooks) is still valid, but it's not required from the perspective that linking on mobile now works much better with WalletConnect.
